### PR TITLE
ham/eng 266 global wallet with rainbow wagmi

### PR DESCRIPTION
**Pull request by ham-evans**
- added rainbow + wagmi, custom theme, added avax
- fixing imports
- mint button component
- fixing mint button
- fixing errors, wagmi hook
- fixing contract bug
- handling invalid chains, error messages
- fixing chain + ui bugs
- loading fix
- packs update
- for pack NFT's can claim on different chain than they have NFT's on
- reverting last changes due to it breaking
- updated contracts
- adding optimism
- use theme merge, adjust header spacing and button styles
- updated packs and claim page
- cleaning up constants
- packs + claim update
- fixing tx link, getting tx error
- fixing tx link
- moving useeffect helpers to lib/fetchers
- fixing moving image
- use clsx
- use clsx
- chain is valid bug

### CODE REVIEW

**Comment by illepic** [comment]
I've converted this to draft and am working on tweaks.

File: pages/_app.tsx [hunk 1]
```diff
@@ -3,13 +3,63 @@ import Router from "next/router";
...
+    body: "Poppins, sans-serif",
+  },
+  radii: {
+    connectButton: "4px",
+    modal: "8px",
+    modalMobile: "8px",
+  },
+} as Theme);
+
+export const { chains, provider } = configureChains(
```
Comment (illepic) [inline pages/_app.tsx:43]: Why are we exporting here and below?

File: pages/_app.tsx [hunk 2]
```diff
@@ -3,13 +3,63 @@ import Router from "next/router";
...
+    jsonRpcProvider({ rpc: (chain) => ({ http: chain.rpcUrls.default }) }),
+  ]
+);
+
+const { connectors } = getDefaultWallets({
+  appName: "Infinity Keys",
+  chains,
+});
+
+export const wagmiClient = createClient({
```
Comment (illepic) [inline pages/_app.tsx:57]: Why export here?

File: pages/_app.tsx (hunk 1 above)
Comment (illepic) [inline pages/_app.tsx:43]: I do not see `provider` used within this file?

File: pages/_app.tsx [hunk 3]
```diff
@@ -3,13 +3,63 @@ import Router from "next/router";
...
+  },
+  fonts: {
+    body: "Poppins, sans-serif",
+  },
+  radii: {
+    connectButton: "4px",
+    modal: "8px",
+    modalMobile: "8px",
+  },
+} as Theme);
```
Comment (illepic) [inline pages/_app.tsx]: Can most of these logic-only configurations be put into a separate lib and just imported here?

File: pages/api/minter/check-claimed.ts [hunk 4]
```diff
@@ -15,17 +15,19 @@ export default async function handler(
...
+  // let claimed = false;
+  // for (let i = 0; i < chainIds.length; i++) {
+  //   const contract = contractLookup[chainIds[i]];
+  //   if (await contract.checkIfClaimed(tokenId, account)) claimed = true;
+  // }
+  const contractPromises = chainIds.map((chainId) =>
+    contractLookup[chainId].checkIfClaimed(tokenId, account)
+  );
+  const contractClaims = await Promise.all(contractPromises);
+  const claimed = contractClaims.some(Boolean);
```
Comment (ham-evans) [inline pages/api/minter/check-claimed.ts:30]: Yo this is crazy.. no clue you could do this

File: components/minter.tsx [hunk 5]
```diff
@@ -71,6 +72,7 @@ export default function Minter({ tokenId, gatedIds }: MinterParams) {
 
   // isVerifying = Verify + CheckIfOwned API Calls
   // isLoading = Tx is processing
+  // @TODO: move this to good ol' fashioned if else statements. Or xstate.
```
Comment (ham-evans) [inline components/minter.tsx:96]: yea this was biggest issue with PR i thought lol didn't know what to do with this logic.  If/Else sounds good though

File: components/minter.tsx [hunk 6]
```diff
@@ -90,6 +92,7 @@ export default function Minter({ tokenId, gatedIds }: MinterParams) {
     : "Connect Wallet To Claim Trophy";
 
   const buttonMint = (
+    // @TOOD: utilize existing Button component for this
```
Comment (ham-evans) [inline components/minter.tsx:120]: 👍🏼

**Review by illepic — APPROVED** [review]

**Comment by illepic** [comment]
Approved, but let's hit these comments on a clean-up round.

File: pages/_app.tsx (hunk 1 above)
Comment (ham-evans) [reply pages/_app.tsx:43]: line 60

File: pages/_app.tsx (hunk 3 above)
Comment (ham-evans) [reply pages/_app.tsx]: Nope- tried this, when every build was failing this was the issue.  No clue why but it needs to be in app.tsx directly

**Pull request merged**